src/libmongoc/tests/test-mongoc-handshake.c

```diff
@@ -276,6 +276,122 @@ test_mongoc_handshake_data_append_success (void)
 }
 
 
+static void
+test_mongoc_handshake_data_append_null_args (void)
+{
+   mock_server_t *server;
+   mongoc_uri_t *uri;
+   mongoc_client_t *client;
+   mongoc_client_pool_t *pool;
+   request_t *request;
+   const bson_t *request_doc;
+   bson_iter_t iter;
+   bson_iter_t md_iter;
+   bson_iter_t inner_iter;
+   const char *val;
+
+   char big_string[HANDSHAKE_MAX_SIZE];
+
+   memset (big_string, 'a', HANDSHAKE_MAX_SIZE - 1);
```

Comment:
note: Ok, fill /up to/ the last byte, then the next line, then NULL terminate. (I'm not sure if newer C standards have nicer ways to do this; I believe in C99 you can use a designated initializer for the last byte, and get away with something like:
big_string[HANDSHAKE_MAX_SIZE] = { [0... HANDSHAKE_MAX_SIZE - 1] 'a', NULL };
...but I'm not entirely sure if we're actually supporting C99 yet.

Reply:
I have not looked too deeply into the contents of this test function (which was modelled after the existing `test_mongoc_handshake_data_append_success()` test function).